### DISCUSSION TITLE
Redirect own flows on MacOS

### DISCUSSION THREAD

**Discussion by emanuele-em** [tinyproxy/tinyproxy#486] [Q&A]
Hi everyone, I would like to understand how TinyProxy handles redirecting traffic that is coming from the same machine where the proxy is running, and how it avoids potential loops on macOS. I have searched for documentation on this, but have not found anything conclusive.

Could someone provide an explanation of how this works in tinyproxy? Does it involve the NetworkExtensionProvider framework?

**Comment by rofl0r** [comment]
detecting loops was implemented in cd005a94cec38e73ca796f1d142c193f48aaa27f . we don't use any MacOS specific functionality, but POSIX standard functionality.

**Reply by emanuele-em** [reply]
thank you for your reply but Can Tinyproxy redirect HTTPS traffic transparently from the machine where it is launched?

**Reply by rofl0r** [reply]
depends on what you mean with redirect and with transparently.

**Reply by emanuele-em** [reply]
I want to intercept traffic from the same machine where tinyproxy is launched without vm or docker, and I want to do it withouth explicitly set the proxy on the client (that is the same machine where tinyproxy is running)

**Reply by rofl0r** [reply]
tinyproxy supports "transparent proxy" mode, however in the age of https this is only of limited usefulness. as soon as TLS comes into play, dealing with certificates becomes a mess.
